scripts/reportportal/rp_manual_reporter/rp_manual_reporter.py

```
@@ -0,0 +1,764 @@
+        "CHANNEL": "--channel",
+    }
+    if not dry_run:
+        present_keys = {attr["key"] for attr in attributes}
+        missing = {key: flag for key, flag in required_attr_keys.items() if key not in present_keys}
+        if missing:
+            missing_names = ", ".join(sorted(missing.keys()))
+            missing_flags = ", ".join(sorted(missing.values()))
+            raise click.ClickException(
+                f"Missing required launch attributes: {missing_names}\n"
+                f"Provide them via CLI flags ({missing_flags}) or use --from-cluster."
+            )
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)